### PR TITLE
log lvl change in cache query

### PR DESCRIPTION
**Pull request by hieunguyenmsft**
## Proposed changes

Reduce noises in log when query cache

Example:

> 2021-10-28T22:12:05.784Z [I-AH][MSAL] TID=3775080 MSAL 1.1.21 iOS 14.8 [2021-10-28 22:12:05 - 1D466778-3346-4A4F-BCAA-5730E6D1528C] [MSAL] (MSIDAccountCredentialCache) retrieved 31 cached credentials  TSMSALAuthProvider:896
2021-10-28T22:12:05.784Z [I-AH][MSAL] TID=3775080 MSAL 1.1.21 iOS 14.8 [2021-10-28 22:12:05 - 1D466778-3346-4A4F-BCAA-5730E6D1528C] [MSAL] (MSIDAccountCredentialCache) credential query requires exact match with the cached credential items. Performing additional filtering checks.  TSMSALAuthProvider:896
2021-10-28T22:12:05.784Z [I-AH][MSAL] TID=3775080 MSAL 1.1.21 iOS 14.8 [2021-10-28 22:12:05 - 1D466778-3346-4A4F-BCAA-5730E6D1528C] [MSAL] (MSIDAccountCredentialCache) cached item had mismatching realm/clientId/familyId/target/requestedClaims with the credential query. excluding from the results.  TSMSALAuthProvider:896
2021-10-28T22:12:05.784Z [I-AH][MSAL] TID=3775080 MSAL 1.1.21 iOS 14.8 [2021-10-28 22:12:05 - 1D466778-3346-4A4F-BCAA-5730E6D1528C] [MSAL] (MSIDAccountCredentialCache) cached item had mismatching realm/clientId/familyId/target/requestedClaims with the credential query. excluding from the results.  TSMSALAuthProvider:896
2021-10-28T22:12:05.784Z [I-AH][MSAL] TID=3775080 MSAL 1.1.21 iOS 14.8 [2021-10-28 22:12:05 - 1D466778-3346-4A4F-BCAA-5730E6D1528C] [MSAL] (MSIDAccountCredentialCache) cached item had mismatching realm/clientId/familyId/target/requestedClaims with the credential query. excluding from the results.  TSMSALAuthProvider:896
2021-10-28T22:12:05.784Z [I-AH][MSAL] TID=3775080 MSAL 1.1.21 iOS 14.8 [2021-10-28 22:12:05 - 1D466778-3346-4A4F-BCAA-5730E6D1528C] [MSAL] (MSIDAccountCredentialCache) cached item had mismatching realm/clientId/familyId/target/requestedClaims with the credential query. excluding from the results.  TSMSALAuthProvider:896
2021-10-28T22:12:05.785Z [I-AH][MSAL] TID=3775080 MSAL 1.1.21 iOS 14.8 [2021-10-28 22:12:05 - 1D466778-3346-4A4F-BCAA-5730E6D1528C] [MSAL] (MSIDAccountCredentialCache) cached item had mismatching realm/clientId/familyId/target/requestedClaims with the credential query. excluding from the results.  TSMSALAuthProvider:896
2021-10-28T22:12:05.785Z [I-AH][MSAL] TID=3775080 MSAL 1.1.21 iOS 14.8 [2021-10-28 22:12:05 - 1D466778-3346-4A4F-BCAA-5730E6D1528C] [MSAL] (MSIDAccountCredentialCache) cached item had mismatching realm/clientId/familyId/target/requestedClaims with the credential query. excluding from the results.  TSMSALAuthProvider:896
2021-10-28T22:12:05.785Z [I-AH][MSAL] TID=3775080 MSAL 1.1.21 iOS 14.8 [2021-10-28 22:12:05 - 1D466778-3346-4A4F-BCAA-5730E6D1528C] [MSAL] (MSIDAccountCredentialCache) cached item had mismatching realm/clientId/familyId/target/requestedClaims with the credential query. excluding from the results.  TSMSALAuthProvider:896
2021-10-28T22:12:05.785Z [I-AH][MSAL] TID=3775080 MSAL 1.1.21 iOS 14.8 [2021-10-28 22:12:05 - 1D466778-3346-4A4F-BCAA-5730E6D1528C] [MSAL] (MSIDAccountCredentialCache) cached item had mismatching realm/clientId/familyId/target/requestedClaims with the credential query. excluding from the results.  TSMSALAuthProvider:896
2021-10-28T22:12:05.785Z [I-AH][MSAL] TID=3775080 MSAL 1.1.21 iOS 14.8 [2021-10-28 22:12:05 - 1D466778-3346-4A4F-BCAA-5730E6D1528C] [MSAL] (MSIDAccountCredentialCache) cached item had mismatching realm/clientId/familyId/target/requestedClaims with the credential query. excluding from the results.  TSMSALAuthProvider:896
2021-10-28T22:12:05.785Z [I-AH][MSAL] TID=3775080 MSAL 1.1.21 iOS 14.8 [2021-10-28 22:12:05 - 1D466778-3346-4A4F-BCAA-5730E6D1528C] [MSAL] (MSIDAccountCredentialCache) cached item had mismatching realm/clientId/familyId/target/requestedClaims with the credential query. excluding from the results.  TSMSALAuthProvider:896
2021-10-28T22:12:05.785Z [I-AH][MSAL] TID=3775080 MSAL 1.1.21 iOS 14.8 [2021-10-28 22:12:05 - 1D466778-3346-4A4F-BCAA-5730E6D1528C] [MSAL] (MSIDAccountCredentialCache) cached item had mismatching realm/clientId/familyId/target/requestedClaims with the credential query. excluding from the results.  TSMSALAuthProvider:896
2021-10-28T22:12:05.785Z [I-AH][MSAL] TID=3775080 MSAL 1.1.21 iOS 14.8 [2021-10-28 22:12:05 - 1D466778-3346-4A4F-BCAA-5730E6D1528C] [MSAL] (MSIDAccountCredentialCache) cached item had mismatching realm/clientId/familyId/target/requestedClaims with the credential query. excluding from the results.  TSMSALAuthProvider:896
2021-10-28T22:12:05.785Z [I-AH][MSAL] TID=3775080 MSAL 1.1.21 iOS 14.8 [2021-10-28 22:12:05 - 1D466778-3346-4A4F-BCAA-5730E6D1528C] [MSAL] (MSIDAccountCredentialCache) cached item had mismatching realm/clientId/familyId/target/requestedClaims with the credential query. excluding from the results.  TSMSALAuthProvider:896
2021-10-28T22:12:05.785Z [I-AH][MSAL] TID=3775080 MSAL 1.1.21 iOS 14.8 [2021-10-28 22:12:05 - 1D466778-3346-4A4F-BCAA-5730E6D1528C] [MSAL] (MSIDAccountCredentialCache) cached item had mismatching realm/clientId/familyId/target/requestedClaims with the credential query. excluding from the results.  TSMSALAuthProvider:896
2021-10-28T22:12:05.785Z [I-AH][MSAL] TID=3775080 MSAL 1.1.21 iOS 14.8 [2021-10-28 22:12:05 - 1D466778-3346-4A4F-BCAA-5730E6D1528C] [MSAL] (MSIDAccountCredentialCache) cached item had mismatching realm/clientId/familyId/target/requestedClaims with the credential query. excluding from the results.  TSMSALAuthProvider:896
2021-10-28T22:12:05.786Z [I-AH][MSAL] TID=3775080 MSAL 1.1.21 iOS 14.8 [2021-10-28 22:12:05 - 1D466778-3346-4A4F-BCAA-5730E6D1528C] [MSAL] (MSIDAccountCredentialCache) cached item had mismatching realm/clientId/familyId/target/requestedClaims with the credential query. excluding from the results.  TSMSALAuthProvider:896
2021-10-28T22:12:05.786Z [I-AH][MSAL] TID=3775080 MSAL 1.1.21 iOS 14.8 [2021-10-28 22:12:05 - 1D466778-3346-4A4F-BCAA-5730E6D1528C] [MSAL] (MSIDAccountCredentialCache) cached item had mismatching realm/clientId/familyId/target/requestedClaims with the credential query. excluding from the results.  TSMSALAuthProvider:896
2021-10-28T22:12:05.786Z [I-AH][MSAL] TID=3775080 MSAL 1.1.21 iOS 14.8 [2021-10-28 22:12:05 - 1D466778-3346-4A4F-BCAA-5730E6D1528C] [MSAL] (MSIDAccountCredentialCache) cached item had mismatching realm/clientId/familyId/target/requestedClaims with the credential query. excluding from the results.  TSMSALAuthProvider:896
2021-10-28T22:12:05.786Z [I-AH][MSAL] TID=3775080 MSAL 1.1.21 iOS 14.8 [2021-10-28 22:12:05 - 1D466778-3346-4A4F-BCAA-5730E6D1528C] [MSAL] (MSIDAccountCredentialCache) cached item had mismatching realm/clientId/familyId/target/requestedClaims with the credential query. excluding from the results.  TSMSALAuthProvider:896
2021-10-28T22:12:05.786Z [I-AH][MSAL] TID=3775080 MSAL 1.1.21 iOS 14.8 [2021-10-28 22:12:05 - 1D466778-3346-4A4F-BCAA-5730E6D1528C] [MSAL] (MSIDAccountCredentialCache) cached item had mismatching realm/clientId/familyId/target/requestedClaims with the credential query. excluding from the results.  TSMSALAuthProvider:896
2021-10-28T22:12:05.786Z [I-AH][MSAL] TID=3775080 MSAL 1.1.21 iOS 14.8 [2021-10-28 22:12:05 - 1D466778-3346-4A4F-BCAA-5730E6D1528C] [MSAL] (MSIDAccountCredentialCache) cached item had mismatching realm/clientId/familyId/target/requestedClaims with the credential query. excluding from the results.  TSMSALAuthProvider:896
2021-10-28T22:12:05.786Z [I-AH][MSAL] TID=3775080 MSAL 1.1.21 iOS 14.8 [2021-10-28 22:12:05 - 1D466778-3346-4A4F-BCAA-5730E6D1528C] [MSAL] (MSIDAccountCredentialCache) cached item had mismatching realm/clientId/familyId/target/requestedClaims with the credential query. excluding from the results.  TSMSALAuthProvider:896
2021-10-28T22:12:05.786Z [I-AH][MSAL] TID=3775080 MSAL 1.1.21 iOS 14.8 [2021-10-28 22:12:05 - 1D466778-3346-4A4F-BCAA-5730E6D1528C] [MSAL] (MSIDAccountCredentialCache) cached item had mismatching realm/clientId/familyId/target/requestedClaims with the credential query. excluding from the results.  TSMSALAuthProvider:896
2021-10-28T22:12:05.786Z [I-AH][MSAL] TID=3775080 MSAL 1.1.21 iOS 14.8 [2021-10-28 22:12:05 - 1D466778-3346-4A4F-BCAA-5730E6D1528C] [MSAL] (MSIDAccountCredentialCache) cached item had mismatching realm/clientId/familyId/target/requestedClaims with the credential query. excluding from the results.  TSMSALAuthProvider:896
2021-10-28T22:12:05.786Z [I-AH][MSAL] TID=3775080 MSAL 1.1.21 iOS 14.8 [2021-10-28 22:12:05 - 1D466778-3346-4A4F-BCAA-5730E6D1528C] [MSAL] (MSIDAccountCredentialCache) cached item had mismatching realm/clientId/familyId/target/requestedClaims with the credential query. excluding from the results.  TSMSALAuthProvider:896
2021-10-28T22:12:05.786Z [I-AH][MSAL] TID=3775080 MSAL 1.1.21 iOS 14.8 [2021-10-28 22:12:05 - 1D466778-3346-4A4F-BCAA-5730E6D1528C] [MSAL] (MSIDAccountCredentialCache) cached item had mismatching realm/clientId/familyId/target/requestedClaims with the credential query. excluding from the results.  TSMSALAuthProvider:896
2021-10-28T22:12:05.786Z [I-AH][MSAL] TID=3775080 MSAL 1.1.21 iOS 14.8 [2021-10-28 22:12:05 - 1D466778-3346-4A4F-BCAA-5730E6D1528C] [MSAL] (MSIDAccountCredentialCache) cached item had mismatching realm/clientId/familyId/target/requestedClaims with the credential query. excluding from the results.  TSMSALAuthProvider:896
2021-10-28T22:12:05.786Z [I-AH][MSAL] TID=3775080 MSAL 1.1.21 iOS 14.8 [2021-10-28 22:12:05 - 1D466778-3346-4A4F-BCAA-5730E6D1528C] [MSAL] (MSIDAccountCredentialCache) cached item had mismatching realm/clientId/familyId/target/requestedClaims with the credential query. excluding from the results.  TSMSALAuthProvider:896
2021-10-28T22:12:05.786Z [I-AH][MSAL] TID=3775080 MSAL 1.1.21 iOS 14.8 [2021-10-28 22:12:05 - 1D466778-3346-4A4F-BCAA-5730E6D1528C] [MSAL] (MSIDAccountCredentialCache) cached item had mismatching realm/clientId/familyId/target/requestedClaims with the credential query. excluding from the results.  TSMSALAuthProvider:896
2021-10-28T22:12:05.786Z [I-AH][MSAL] TID=3775080 MSAL 1.1.21 iOS 14.8 [2021-10-28 22:12:05 - 1D466778-3346-4A4F-BCAA-5730E6D1528C] [MSAL] (MSIDAccountCredentialCache) cached item had mismatching realm/clientId/familyId/target/requestedClaims with the credential query. excluding from the results.  TSMSALAuthProvider:896
2021-10-28T22:12:05.786Z [I-AH][MSAL] TID=3775080 MSAL 1.1.21 iOS 14.8 [2021-10-28 22:12:05 - 1D466778-3346-4A4F-BCAA-5730E6D1528C] [MSAL] (MSIDAccountCredentialCache) returning 1 filtered credentials  TSMSALAuthProvider:896

We have 31 cache items, filter out most of them and eventually return 1. There is no error in this scenario so I think we can reduce the noise in log.

## Type of change

- [ ] Feature work
- [ ] Bug fix
- [ ] Documentation
- [ ] Engineering change
- [ ] Test
- [x] Logging/Telemetry

## Risk

- [ ] High – Errors could cause MAJOR regression of many scenarios. (Example: new large features or high level infrastructure changes)
- [ ] Medium – Errors could cause regression of 1 or more scenarios. (Example: somewhat complex bug fixes, small new features)
- [ ] Small – No issues are expected. (Example: Very small bug fixes, string changes, or configuration settings changes)

### CODE REVIEW

File: IdentityCore/src/cache/crypto/MSIDAssymetricKeyPair.m [hunk 1]
```diff
@@ -92,7 +92,7 @@ - (NSString *)keyExponent
         iterator++; // TYPE - bit stream exp
         int exp_size = [self derEncodingGetSizeFrom:publicKeyBits at:&iterator];
         
-        _keyExponent = [[publicKeyBits subdataWithRange:NSMakeRange(iterator, exp_size)] base64EncodedStringWithOptions:0];
+        _keyExponent = [[publicKeyBits subdataWithRange:NSMakeRange(iterator, exp_size)] msidBase64UrlEncodedString];
```
Comment (oldalton) [inline IdentityCore/src/cache/crypto/MSIDAssymetricKeyPair.m:95]: This change seems unrelated, can you merge in latest dev?

**Review by oldalton — APPROVED** [review]

**Review by antrix1989 — APPROVED** [review]
